src/components/FollowAllButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect, useContext } from 'react';
+import Button from '@/components/Button';
+import { useConnection, useAnchorWallet } from '@solana/wallet-adapter-react';
+
+import { Program } from '@holaplex/graph-program';
+import * as anchor from '@project-serum/anchor';
+import { Action, MultiTransactionContext } from 'src/views/_global/MultiTransaction';
+import { shortenAddress } from '@/modules/utils/string';
+import { useConnectedWalletProfile } from '../views/_global/ConnectedWalletProfileProvider';
+import { Transaction } from '@solana/web3.js';
+import { useRouter } from 'next/router';
+
+export interface ProfileToFollow {
+  address: string;
+  handle?: string;
+  profileImageUrl?: string;
+}
+
+const FollowAllButton = (props: { profilesToFollow?: ProfileToFollow[] }) => {
+  const { connection } = useConnection();
+  const anchorWallet = useAnchorWallet();
+  const router = useRouter();
+  const myPubkey = anchorWallet?.publicKey.toBase58() || '';
+
+  const { connectedProfile } = useConnectedWalletProfile();
+  const myFollowingList: string[] | undefined = connectedProfile?.following?.map((f) => f.address);
+
+  const [actualProfilesToFollow, setactualProfilesToFollow] = useState<ProfileToFollow[]>([]);
+
+  useEffect(() => {
+    if (props.profilesToFollow) {
+      setactualProfilesToFollow(
+        props.profilesToFollow.filter((u) => !myFollowingList?.includes(u.address))
+      );
+    }
+  }, [myFollowingList, myFollowingList?.length, props.profilesToFollow]);
```

Comment:
Can the profiles the current user is following be filtered out when passed to this component instead of using state.